app/org/sagebionetworks/bridge/services/ConsentService.java

```diff
@@ -114,80 +118,77 @@ public ConsentSignature getConsentSignature(Study study, SubpopulationGuid subpo
     }
     
     /**
-     * Consent this user to research. User will be updated to reflect consent.
+     * Consent this user to research. User will be updated to reflect consent. This method will ensure the 
+     * user is not already consented to this subpopulation, but it does not validate that the user is a 
+     * validate member of this subpopulation (that is checked in the controller).
+     * 
      * @param study
      * @param subpopGuid
-     * @param session
+     * @param participant
      * @param consentSignature
      * @param sharingScope
      * @param sendEmail
+     *      if true, send the consent document to the user's email address
      * @return
+     * @throws EntityNotFoundException
+     *      if the subpopulation is not part of the study
+     * @throws InvalidEntityException
+     *      if the user is not old enough to participate in the study (based on birthdate declared in signature)
      * @throws EntityAlreadyExistsException
-     *      if the user already has an active consent to participate in research
-     * @throws StudyLimitExceededException
-     *      if enrolling the user would exceed the study enrollment limit
+     *      if the user has already signed the consent for this subpopulation
      */
-    public void consentToResearch(Study study, SubpopulationGuid subpopGuid, UserSession session, ConsentSignature consentSignature, 
-            SharingScope sharingScope, boolean sendEmail) {
+    public void consentToResearch(Study study, SubpopulationGuid subpopGuid, StudyParticipant participant,
+            ConsentSignature consentSignature, SharingScope sharingScope, boolean sendEmail) {
         checkNotNull(study, Validate.CANNOT_BE_NULL, "study");
         checkNotNull(subpopGuid, Validate.CANNOT_BE_NULL, "subpopulationGuid");
-        checkNotNull(session, Validate.CANNOT_BE_NULL, "session");
+        checkNotNull(participant, Validate.CANNOT_BE_NULL, "participant");
         checkNotNull(consentSignature, Validate.CANNOT_BE_NULL, "consentSignature");
         checkNotNull(sharingScope, Validate.CANNOT_BE_NULL, "sharingScope");
 
-        ConsentStatus status = session.getConsentStatuses().get(subpopGuid);
-        // There will be a status object for each subpopulation the user is mapped to. If 
-        // there's no status object, then in effect the subpopulation does not exist for 
-        // this user and they should get back a 404.
-        if (status == null) {
-            throw new EntityNotFoundException(Subpopulation.class);
-        }
-        if (status.isConsented()) {
+        Account account = accountDao.getAccount(study, participant.getId());
+        ConsentSignature active = account.getActiveConsentSignature(subpopGuid);
```

Comment:
This will not allow you to re-consent to a more up-to-date consent version. We should check and only throw exception if the active consent timestamp is later than the signature's.